app/javascript/vue/shared/checklist_select.vue

```diff
@@ -0,0 +1,141 @@
+          <div class="sci-checkbox-container">
+            <input v-model="selectedValues" :value="option.id" :id="option.id" type="checkbox" class="sci-checkbox project-card-selector mr-1">
+            <label :for="option.id" class="sci-checkbox-label"></label>
+          </div>
+          <span class="text-ellipsis overflow-hidden max-h-[4rem]">{{ option.label }}</span>
+        </div>
+      </div>
+      <template v-else>
+        <div
+          class="sn-select__no-options"
+        >
+          {{ this.noOptionsPlaceholder }}
+        </div>
+      </template>
+    </perfect-scrollbar>
+  </div>
+</template>
+
+<script>
+  import PerfectScrollbar from 'vue2-perfect-scrollbar';
+
+  export default {
+    name: 'ChecklistSelect',
+    comments: { PerfectScrollbar },
+    props: {
+      withButtons: { type: Boolean, default: false },
+      withEditCursor: { type: Boolean, default: false },
+      initialSelectedValues: { type: Array, default: () => [] },
+      options: { type: Array, default: () => [] },
+      placeholder: { type: String },
+      noOptionsPlaceholder: { type: String },
+      disabled: { type: Boolean, default: false }
+    },
+    data() {
+      return {
+        selectedValues: [],
+        isOpen: false,
+        optionPositionStyle: ''
+      }
+    },
+    mounted() {
+      this.selectedValues = this.initialSelectedValues;
+    },
+    computed: {
+      valueLabel() {
+        if (!this.selectedValues.length) {
+          return (this.placeholder || this.i18n.t('general.select'));
+        }
+
+        return `${this.selectedValues.length} ${this.i18n.t('general.selected')}`;
+      }
+    },
+    watch: {
+      initialSelectedValues: {
+        handler: function (newVal) {
+          this.selectedValues = newVal;
+        },
+        deep: true
+      }
+    },
+    methods: {
+      toggle() {
+        this.isOpen = !this.isOpen;
+        if (this.isOpen) {
+          this.updateOptionPosition();
+        } else {
+          this.close();
+        }
+      },
+      updateOptionPosition() {
+        const container = $(this.$refs.container);
+        const rect = container.get(0).getBoundingClientRect();
+        let width = rect.width;
+        let height = rect.height;
+        this.optionPositionStyle = `position: absolute; top: ${height}px; left: 0px; width: ${width}px`
+      },
+      toggleOption(id) {
+        if (this.selectedValues.includes(id)) {
+          this.selectedValues = this.selectedValues.filter((value) => value !== id);
+        } else {
+          this.selectedValues.push(id);
+        }
+      },
+      close() {
+        if (!this.isOpen) return;
+
+        this.isOpen = false;
+        if (this.selectedValues.length) {
+          this.$emit('update', this.selectedValues);
+        }
+      },
+      isSelected(id) {
+        return this.selectedValues.includes(id);
+      }
+    }
+  }
+</script>
```

Comment:
Ideally this would use the base Select component (same as search-select does), but this is OK too.

Reply:
Yeah, I thought about how we will handle multiple choices at once